### DISCUSSION TITLE
Feature request: React Native Modules

### DISCUSSION THREAD

**Discussion by SohelIslamImran** [algolia/instantsearch#5863] [Ideas]
Combine [instantsearch-ios](https://github.com/algolia/instantsearch-ios) and [instantsearch-android](https://github.com/algolia/instantsearch-android) and create an [**React Native Modules**](https://reactnative.dev/docs/native-modules-intro) for better performance and capabilities. Also, it would be great if you also combine [algoliasearch-client-swift](https://github.com/algolia/algoliasearch-client-swift) and [algoliasearch-client-kotlin](https://github.com/algolia/algoliasearch-client-kotlin) for React Native.

[Native Modules Intro](https://reactnative.dev/docs/native-modules-intro)
https://reactnative.dev/docs/native-modules-android
https://reactnative.dev/docs/native-modules-ios
https://reactnative.dev/docs/native-modules-setup

Expo Modules is nice and simple way to create 
https://docs.expo.dev/modules/overview/

**Comment by Haroenv** [comment]
Hi @SohelIslamImran, we have no plans to do this, as the architecture of the native iOS and android clients and native InstantSearch is completely different to the architecture of the the web-based InstantSearch flavours.

You can use either:
- React InstantSearch Core for react native, which has been used in production by many applications and has the exact same features as our web flavours
- InstantSearch for iOS and android and create two apps
- create your custom implementation with the javascript client
- create your custom implementation with the swift and kotlin clients and some conditional code
